crates_io_tarball/src/lib.rs

```diff
@@ -93,10 +97,14 @@ pub fn process_tarball<R: Read>(
             // erroring if it cannot be read.
             let mut contents = String::new();
             entry.read_to_string(&mut contents)?;
-            manifest = toml::from_str(&contents).ok();
+            manifest = Some(toml::from_str(&contents).map_err(TarballError::InvalidManifest)?);
```

Comment:
Something we'll have to keep in mind as we add new features is to determine if crates-io's validation code needs updating.

Reply:
(to be clear, atm it is very minimal so its less likely)

Reply:
yep, definitely!